src/DataSourceProvider.vue

```diff
@@ -1,19 +1,367 @@
 <template>
-  <div>
-    <h1>Your fliplet-widget-data-source-provider widget will be added here.</h1>
-    <First-Component></First-Component>
-  </div>
+  <section>
+    <div class="data-source-title">
+      <strong>{{ widgetData.dataSourceTitle || 'Select a data source' }}</strong>
+    </div>
+
+    <div v-if="isLoading" class="spinner-holder animated">
+      <div class="spinner-overlay">Loading...</div>
+    </div>
+
+    <div v-else-if="hasError" class="alert alert-danger" role="alert">
+      {{ errorMessage }}
+    </div>
+
+    <div v-else class="main-data-source-provider">
+      <section class="data-source-selector">
+
+        <div v-if="dataSources.length && changeDataSource">
+
+          <Select2
+            :dataSources="dataSources"
+            :selectedDataSource.sync="selectedDataSource"
+            :customOptionView="formatDataSourceOption"
+            :customSearch="customDataSourceSearch"
+            :optionLabelKey="'name'"
+            :optionValueKey="'id'"
+            :selectWithGroups="!!allDataSources.length"
+          >
+          </Select2>
+
+          <a @click.prevent="onDataSourceCreate" class="create-data-source" href="#">Create new data source</a>
+
+          <div class="checkbox checkbox-icon">
+            <input :checked="showAll" type="checkbox" name="showAll" id="showAll" />
+            <label for="showAll">
+              <span class="check">
+                <i class="fa fa-check"></i>
+              </span>
+              Show all data sources
+            </label>
+          </div>
+
+        </div>
+
+        <div v-else-if="selectedDataSource && !changeDataSource">
+          <p>{{ selectedDataSource.id }}. {{ selectedDataSource.name }} <a @click.prevent="onDataSourceChange" class="change-data-source">Change</a></p>
```

Comment:
```suggestion
          <p><code>{{ selectedDataSource.id }}</code> {{ selectedDataSource.name }} <a @click.prevent="onDataSourceChange" class="change-data-source">Change</a></p>
```

Reply:
@tonytlwu It will look like this. Is that okay?
![image](https://user-images.githubusercontent.com/53430352/92722899-f1647200-f370-11ea-97d1-fc77423a460a.png)

Reply:
@YaroslavOvdii Yup, that works. I realised that the **1.** in the mockup was actually part of the data source name, but your code here reminds me that the data source ID might be useful, hence my suggeestion.

Reply:
@tonytlwu Added these changes to the code.